tools/serverpod_cli/lib/src/commands/upgrade.dart

```diff
@@ -0,0 +1,30 @@
+import 'dart:io';
+
+import 'package:serverpod_cli/src/logger/logger.dart';
+import 'package:serverpod_cli/src/runner/serverpod_command.dart';
+
+import '../generated/version.dart';
+
+class UpgradeCommand extends ServerpodCommand {
+  @override
+  final name = 'upgrade';
+
+  @override
+  final description = 'Upgrade Serverpod to the latest version.';
+
+  @override
+  void run() async {
+    dynamic error;
+    var success = await log.progress('Running `dart pub global activate serverpod_cli`...', () async {
+      var process = await Process.run('dart', ['pub global activate serverpod_clipper']);
+      error = process.stderr;
+      return !process.exitCode.isNegative;
+    });
+
+    if (success) {
+      log.info('Serverpod is already up to date: $templateVersion version.');
```

Comment:
Suggestion: remove `already` as we performed the upgrade.

Reply:
Modified in last commit 😀

Reply:
Chore: This still seems to be the case. Could you make sure that this reporting is correct?

Make sure we print correctly when:
- The update was successful
- The update fails
- The update is a no-op.

Reply:
I understand that when we talk about a no-op, we mean when you already have the latest version and it's not going to perform any operation. For this I could take a token with access to public repositories and connect to the GitHub API asking for the latest Tag and compare it with $templateVersion. But is this what we want? Or do we just let it repeat the process?

Reply:
Maybe we don't need to make the no-op case since it adds complexity to little value. Just removing the `already` as Isak suggested is probably enough to cover both the no-op and the success case.

Reply:
And what do you think about the additional log when it fails that I added in the last commit? Redundant?
<img width="585" alt="image" src="https://github.com/serverpod/serverpod/assets/34163765/02587f0e-7486-477d-abc8-e2ff834809fa">

Reply:
How about `Failed to update Serverpod`? :)

Very NIT but users will be thankful :)

Could also merge latest main into the branch just to ensure that all tests pass with latest main as well.

Reply:
Done 😀